roles/openshift_health_checker/openshift_checks/efk_stack.py

```diff
@@ -5,344 +5,601 @@
 
+    def _get_pods(self, task_vars):
+        '''Get all pods and sort them by EFK component. Returns: pods by component, error string'''
+        pod_output = self._exec_oc("get pods -o json", [], task_vars)
         try:
-            cluster_nodes = json.loads(self.exec_oc(node_cmd, [], task_vars))['nodes']
+            pods = json.loads(pod_output)
+            if not pods or not pods.get('items'):
+                raise ValueError()
         except ValueError:
-            return es_status
+            # successful run but non-parsing data generally means there were no pods in the namespace
+            return None, 'There are no pods in the {} namespace. Is logging deployed?'.format(self.logging_namespace)
 
-        es_status['all_nodes_registered'] = 1
-        # The internal ES node name is a random string we do not track anywhere
-        # pylint: disable=unused-variable
-        for node, data in cluster_nodes.items():
-            has_matched = False
-            for pod in es_pods:
-                if data['host'] == pod['metadata']['name'] or data['host'] == pod['status']['podIP']:
-                    has_matched = True
-                    break
+        logging_components = ['es', 'fluentd', 'kibana', 'curator']
+        pods_for = {comp: [] for comp in logging_components}
 
-            if has_matched is False:
-                es_status['all_nodes_registered'] = 0
+        for pod in pods['items']:
+            component = pod['metadata']['labels'].get('component')
+            if component in logging_components:
+                pods_for[component].append(pod)
 
-        return es_status
+        return pods_for, None
 
-    def check_es_cluster_health(self, es_pod, task_vars):
-        """Exec into the elasticsearch pod and check the cluster health"""
-        try:
-            curl_cmd = "{} 'https://localhost:9200/_cluster/health?pretty=true'".format(self.es_curl)
-            cluster_health = "exec -ti {} -- {}".format(es_pod, curl_cmd)
-            health_res = json.loads(self.exec_oc(cluster_health, [], task_vars))
+    @staticmethod
+    def _not_running_pods(pods):
+        '''Returns: list of pods not in a ready and running state'''
+        return [
+            pod for pod in pods
+            if any(
+                container['ready'] is False
+                for container in pod['status']['containerStatuses']
+            ) or not any(
+                condition['type'] == 'Ready' and condition['status'] == 'True'
+                for condition in pod['status']['conditions']
+            )
+        ]
+
+    def _running_es_pods(self, es_pods):
+        '''Returns: list of running pods, list of errors about non-running pods'''
+        not_running = self._not_running_pods(es_pods)
+        if not_running:
+            return [pod for pod in es_pods if pod not in not_running], [(
+                'The following Elastic Search pods are not running:\n'
```

Comment:
Agreed the user messaging should be consistent. Somewhere I got in my head that "Elastic Search" was what Elastic wanted to call it but I see [it's not](https://www.elastic.co/products/elasticsearch). Thanks.